Bottom: 7bf18695fef78ce1cf020b98bda96e2687917519
Top:    7cce5cae5d42cd31d7c3833ac60d0668c0c1d6d5
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-07 11:08:30 -0700

support for debug prints


---

diff --git a/src/preload.c b/src/preload.c
index 33eb9a3..d1b6cff 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -48,6 +48,7 @@
 #include <netinet/tcp.h>
 #include <unistd.h>
 #include <semaphore.h>
+#include <stdio.h>
 
 #include <rdma/rdma_cma.h>
 #include <rdma/rdma_verbs.h>
@@ -98,6 +99,8 @@ static int rq_size;
 static int sq_inline;
 static int fork_support;
 
+static FILE *fout;
+
 enum fd_type {
 	fd_normal,
 	fd_rsocket,
@@ -286,6 +289,7 @@ static void init_preload(void)
 	rs.getsockopt = dlsym(RTLD_DEFAULT, "rgetsockopt");
 	rs.fcntl = dlsym(RTLD_DEFAULT, "rfcntl");
 
+	fout = fopen("rs-out.txt", "w+");
 	getenv_options();
 	init = 1;
 out:
@@ -964,16 +968,21 @@ int dup2(int oldfd, int newfd)
 	return newfd;
 }
 
-int fstat(int socket, struct stat *buf)
+int __fxstat64(int ver, int socket, struct stat *buf)
 {
 	int fd, ret;
 
+	init_preload();
+	fprintf(fout, "fstat socket %d\n", socket);
 	if (fd_get(socket, &fd) == fd_rsocket) {
 		ret = real.fstat(socket, buf);
+		fprintf(fout, "fstat - rsocket %d\n", ret);
 		if (!ret)
 			buf->st_mode = (buf->st_mode & ~S_IFMT) | __S_IFSOCK;
 	} else {
 		ret = real.fstat(fd, buf);
+		fprintf(fout, "fstat - normal %d\n", ret);
 	}
+	fflush(fout);
 	return ret;
 }
